Declaration of pjsip_auth_deinit_parser() gets caught by -Werror=strict-prototypes

### Describe the bug

pjsip_auth_deinit_parser is declared in sip_auth_parser.h as follows:
```
PJ_DECL(void) pjsip_auth_deinit_parser();
```
Notice the empty parameter list.  Since Asterisk compiles with `-Werror=strict-prototypes`, this gets caught and the compile fails.  Asterisk itself doesn't include sip_auth_parser.h but in commit cc680d21e6aac3943f4846b2ef39b0aedc0178dd, sip_auth_parser.h was added to top level pjsip.h so it's exposed to the world by default now.

There are other places in the code that _would_ fail the strict-prototype test but since pjproject itself isn't compiled with that option it's not an issue.

PR forthcoming.

### Steps to reproduce

This will show the include files that would fail if `-Werror=strict-prototypes` was used.  You can't pass `-Werror=strict-prototypes` in the configure command line because then configure itself would fail. :)
```bash
./configure CFLAGS='-Wstrict-prototypes'
make >/dev/null  2> >(egrep "[.]h:.+Wstrict-prototypes")
```


### PJSIP version

master

### Context

Compiling Asterisk against pjproject with commit cc680d21e6aac3943f4846b2ef39b0aedc0178dd,

### Log, call stack, etc

```shell
In file included from /usr/src/asterisk/asterisk/third-party/pjproject/source/pjsip/include/pjsip.h:51,
                 from pjsip/cli_commands.c:29:
/usr/src/asterisk/asterisk/third-party/pjproject/source/pjsip/include/pjsip/sip_auth_parser.h:47:1: error: function declaration isn’t a prototype [-Werror=strict-prototypes]
   47 | PJ_DECL(void) pjsip_auth_deinit_parser();
      | ^~~~~~~
cc1: all warnings being treated as errors
```

--- a/pjsip/include/pjsip/sip_auth_parser.h
+++ b/pjsip/include/pjsip/sip_auth_parser.h
@@ -44,7 +44,7 @@ PJ_DECL(pj_status_t) pjsip_auth_init_parser(void);
 /**
  * DeInitialize authorization parser module.
  */
-PJ_DECL(void) pjsip_auth_deinit_parser();
+PJ_DECL(void) pjsip_auth_deinit_parser(void);
 
 
 

